src/components/SelectButtonGroup/SelectButton/SelectButton.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import bem from '../../../utils/bem';
+import { ENTER_KEY } from '../../../constants';
+import styles from './SelectButton.scss';
+
+// These props will be passed by the parent <SelectButtonGroup>
+interface InternalProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
+    /** A function to be called if a button is pressed. */
+    onChange?: (value: string) => void;
+    /** If this component a child of a full block with container  */
+    isBlock?: boolean;
+    /** If this button has same width as its siblings */
+    isEqualWidth?: boolean;
+}
+
+export interface Props extends InternalProps {
+    /** the label to be displayed on this button */
+    children: NotEmptyReactNode;
+    /** the value associated with this button */
+    value: string;
+    /** if this button should be in a selected state */
+    isSelected?: boolean;
+    /** the color context to be applied in the selected state */
+    selectedContext?: 'neutral' | 'brand';
+}
+
+const { block } = bem('SelectButton', styles);
+
+export const SelectButton: React.FC<Props> = props => {
+    const {
+        children,
+        value,
+        onChange,
+        isEqualWidth,
+        isBlock,
+        selectedContext,
+        isSelected,
+        ...rest
+    } = props;
+
+    const handleClick = () => {
+        onChange?.(value);
```

Comment:
Cool 👍 . For supporting this feature do we need to setup webpack accordingly, correct? Nice to have it for external search.

Reply:
no need to change webpack. TS loader does it if you have TS >=3.7. I did need to update prettier though, as that got confused. Hence https://github.com/textkernel/oneui/pull/217